tests/cases/events_test.py

```diff
@@ -136,3 +137,28 @@ def callback(event):
             self.assertEqual(events.daemon.eventQueue.qsize(), 0)
             self.assertEqual(self.ctr, 3)
             self.assertEqual(self.responses, ['foo'])
+            events.daemon.stop()
+
+    def testForegroundDaemon(self):
+        oldDaemon = events.daemon
+        events.daemon = events.ForegroundEventsDaemon()
+
+        # Should still be able to call start
+        events.daemon.start()
+
+        def callback(event):
+            self.ctr += 1
+            self.responses = event.responses
+
+        with events.bound('_test.event',  '_test.handler', self._raiseException):
+            with six.assertRaisesRegex(self, Exception, 'Failure condition'):
+                events.daemon.trigger('_test.event', None, callback)
+
+        with events.bound('_test.event',  '_test.handler', self._incrementWithResponse):
+            events.daemon.trigger('_test.event', {'amount': 2}, callback)
+
+        self.assertEqual(self.ctr, 3)
+        self.assertEqual(self.responses, ['foo'])
+
+        events.daemon.stop()
+        events.daemon = oldDaemon
```

Comment:
Resetting state at the end makes me nervous.  Can you do the above testing work in a context manager and then reset the state outside the CM?